kotlin-mbedtls/src/main/kotlin/org/opencoap/ssl/SslContext.kt

```diff
@@ -117,6 +117,10 @@ class SslSession internal constructor(
     val ownCid: ByteArray? = if (peerCid != null) cid else null
     val peerCertificateSubject: String? = readPeerCertificateSubject()
 
+    init {
+        require(!conf.requireCid || peerCid != null) { "Peer has to provide CID." }
```

Comment:
We need to think if throwing `IllegalArgumentException` is good here. I would lean to use `SslException` but still we need to see how server will behave to it. So please add unit test like: `SslContextTest#should handshake with certificate` and another test in netty module.

Reply:
Ok.